hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java

```diff
@@ -441,6 +441,9 @@ public String toString() {
 
   @Override
   public long getEstimatedKeyCount() throws IOException {
+    if (cache instanceof FullTableCache) {
```

Comment:
I'd prefer adding `CacheType type()` in `TableCache`, and using `cache.type() == FULL_CACHE` instead of `instanceof` check.